File: push_sale_mobile-master/push_sale_mobile-master/lib/models/address.dart
import 'package:push_sale/models/city.dart';
import 'package:push_sale/models/country.dart';
import 'package:push_sale/models/wilaya.dart';

class Address {
  final String id;
  final String street;
  final String commune;
  final String zipcode;
  final double latitude;
  final double longitude;
  final City city;
  final Wilaya wilaya;
  final Country? country;

  Address({
    required this.id,
    required this.street,
    required this.commune,
    required this.zipcode,
    required this.latitude,
    required this.longitude,
    required this.city,
    required this.wilaya,
    this.country,
  });

  static Address fromMap(Map<String, dynamic> value) {
    return Address(
      id: value["id"],
      street: value["street"] == null ? "" : value["street"],
      commune: value["commune"] == null ? "" : value["commune"],
      zipcode: value["zipcode"] == null ? "" : value["zipcode"],
      latitude: double.parse((value["latitude"] ?? "0").toString()),
      longitude: double.parse((value["longitude"] ?? "0").toString()),
      city: City.fromMap(value["city"]),
      wilaya: Wilaya.fromMap(value["state"]),
      country:
          value["country"] != null ? Country.fromMap(value["country"]) : null,
    );
  }
}

File: push_sale-master/app/Models/Address.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Laravel\Passport\HasApiTokens;


class Address extends Model
{
    use HasApiTokens, HasFactory;
    protected $table = "address";

    public $incrementing = false;
    protected $keyType = 'string';

    protected $fillable = ["id","street", "commune", "zipcode", "latitude", "longitude", "city_id", "state_id", "country_id"];

    protected $hidden = ["state_id", "city_id", "country_id", "created_at", "updated_at"];

    public function city()
    {
        return $this->belongsTo(City::class);
    }
    public function state()
    {
        return $this->belongsTo(State::class);
    }
    public function country()
    {
        return $this->belongsTo(Country::class);
    }
}
